src/points/selectors.ts

```diff
@@ -31,41 +31,13 @@ export const pointsConfigStatusSelector = (state: RootState) => state.points.poi
 
 const pointsConfigSelector = (state: RootState) => state.points.pointsConfig
 
-export const pointsSectionsSelector = createSelector(
-  [pointsConfigSelector],
-  (pointsConfig): PointsMetadata[] => {
-    const pointsMetadata: PointsMetadata[] = []
-
-    Object.entries(pointsConfig.activitiesById).forEach(
-      ([activityId, { pointsAmount } = { pointsAmount: 0 }]) => {
-        if (!isPointsActivityId(activityId)) {
-          // should never happen but Object.entries seems to lose the type for activityId
-          return
-        }
-
-        // check if there is already a metadata object for this points value,
-        // either add activity to the existing points object or create a new one
-        const existingMetadata = pointsMetadata.find(
-          (metadata) => metadata.pointsAmount === pointsAmount
-        )
-        if (existingMetadata) {
-          existingMetadata.activities.push({ activityId })
-        } else {
-          pointsMetadata.push({
-            pointsAmount,
-            activities: [{ activityId }],
-          })
-        }
-      }
-    )
-
-    return pointsMetadata.sort((a, b) => {
-      if (a.pointsAmount < b.pointsAmount) return 1
-      if (a.pointsAmount > b.pointsAmount) return -1
-      return 0
-    })
-  }
-)
+export const pointsActivitiesSelector = createSelector([pointsConfigSelector], (pointsConfig) => {
+  return Object.entries(pointsConfig.activitiesById).map(([activityId, metadata]) => ({
+    ...metadata,
+    activityId,
+    completed: activityId === 'create-wallet',
```

Comment:
should we add the `trackOnceActivitiesSelector` to this selector and use that for this `completed` condition? it doesn't matter so much here but it seems like something we could overlook when adding the next binary activity

Reply:
absolutely! thanks for the suggestion.